### PR TITLE
Allow use of array index with struct pointers

### PR DESCRIPTION
**Pull request by HPCguy**
Bug fix.

### CODE REVIEW

**Comment by HPCguy** [comment]
Closes #83

**Comment by jserv** [comment]
Hint: You can automatically close certain issue(s) within git commit message:. See https://stackoverflow.com/questions/3547445/closing-multiple-issues-in-github-with-a-commit-message

File: amacc.c [hunk 1]
```diff
@@ -983,6 +985,7 @@ void gen(int *n)
     case Load:
         gen(n + 2); // load the value
         if (n[1] <= INT || n[1] >= PTR) { *++e = (n[1] == CHAR) ? LC : LI; }
```
Comment (jserv) [inline amacc.c:987]: Rewrite as following:
```cpp
if (n[1] > INT && n[1] < PTR) fatal("AMaCC does not yet allow whole struct memory access");
*++e = (n[1] == CHAR) ? LC : LI;
break;
```

**Comment by HPCguy** [comment]
Don't commit yet.  I was strengthening tests/struct.c on my local branch and uncovered a bug in the For statement implementation and a hole in the struct array indexing implementation.

**Comment by HPCguy** [comment]
The "init" and "increment" sections of a for-statement are supposed to be optional.  While strengthening tests/struct.c, I found that they were not optional and corrected the oversight.

**Pull request merged**